rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,849 @@
+#
```

Comment:
Will we also have an error code like 500 for an unexpected exception on server side?

Reply:
I would personally choose to leave 5xx off of the document. I consider a potential 500 as an implicit response for any HTTP request, as it's possible the request returns 5xx before even reaching the application server.

I personally try to avoid explicitly returning 5xx as part of the normal flow of code, but implicitly all manner of 5xx should be expected. I'm just not sure if it's worth spelling out for every route.

If others think we should include them, I'd be ok with that.

Reply:
I did some research for this, and in RFC 7231 (https://datatracker.ietf.org/doc/html/rfc7231#section-6.6):

> The 5xx (Server Error) class of status code indicates that the server
   is aware that it has erred or is incapable of performing the
   requested method.  Except when responding to a HEAD request, the
   server SHOULD send a representation containing an explanation of the
   error situation, and whether it is a temporary or permanent

So I think it's better to define that.

Reply:
I see here, in the Confluent Kafka REST open api doc, that they have a `5xx` response. This makes sense to me given that 5xx is often thrown by the server and usually is used that way: https://github.com/confluentinc/kafka-rest/blob/master/api/v3/openapi.yaml#L77-L78

We can iterate on that if we need specific 5xx errors, but I think that's sufficient for now.

As a side note, I am likely going to reorganize some of this document in the style they use, but in a follow up PR. We can discuss then which parts we would like to organize around and which we wouldn't. For now though, I'd like to keep it as close to what we agreed on just to be able to move forward. There will be more PRs opened against this for sure 👍 

But I've added 5xx. Will be adding 400 as well. Thank you both for the review!